src/redisraft.c

```diff
@@ -645,6 +661,72 @@ static void handleRedisCommand(RedisRaftCtx *rr,
     raft_entry_release(entry);
 }
 
+static void handleRedisCommand(RedisRaftCtx *rr,
+                               RedisModuleCtx *ctx, RaftRedisCommandArray *cmds) {
+    /* Handle intercepted commands. We do this also on non-leader nodes or if we
+     * don't have a leader, so it's up to the commands to check these conditions
+     * if they have to.
+     */
+    if (handleInterceptedCommands(rr, ctx, cmds)) {
+        return;
+    }
+
+    /* update cmds array with the client's saved "asking" state */
+    cmds->asking = getAskingState(rr, ctx);
+
+    /* Check if MULTI/EXEC bundling is required. */
+    if (MultiHandleCommand(rr, ctx, cmds)) {
+        return;
+    }
+
+    handleRedisCommandAppend(rr, ctx, cmds);
+
+    /* reset asking state when after executing the command */
+    if (cmds->asking) {
+        setAskingState(rr, ctx, false);
+    }
+}
```

Comment:
```suggestion
    /* update cmds array with the client's saved "asking" state */
    cmds->asking = getAskingState(rr, ctx);
    setAskingState(rr, ctx, false);

    if (handleInterceptedCommands(rr, ctx, cmds)) {
        return;
    }

    /* Check if MULTI/EXEC bundling is required. */
    if (MultiHandleCommand(rr, ctx, cmds)) {
        return;
    }

    handleRedisCommandAppend(rr, ctx, cmds);
}
```

I guess we should clear "asking" state no matter what.